### PR TITLE
Bump version to v0.1.12

### PR DESCRIPTION
**Pull request by charliermarsh**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,57 @@
...
+- [\`flake8-simplify\`] Implement `zip-dict-keys-and-values` (`SIM911`) ([#9460](https://github.com/astral-sh/ruff/pull/9460))
+- [\`pyflakes\`] Add a fix for `redefined-while-unused` (`F811`) ([#9419](https://github.com/astral-sh/ruff/pull/9419))
+- [\`pylint\`] Implement `unnecessary-dunder-call` (`C2801`) ([#9166](https://github.com/astral-sh/ruff/pull/9166))
+- [\`ruff\`] Add `parenthesize-changed-operators` (`RUF021`) to enforce parentheses in `a or b and c` ([#9440](https://github.com/astral-sh/ruff/pull/9440))
+
+### Rule changes
+
+- [\`flake8-boolean-trap\`] Allow Boolean positionals in setters ([#9429](https://github.com/astral-sh/ruff/pull/9429))
+- [\`flake8-bugbear\`] Include subscripts and attributes in static key rule ([#9416](https://github.com/astral-sh/ruff/pull/9416))
+- [\`flake8-bugbear\`] Support variable keys in static dictionary key rule ([#9411](https://github.com/astral-sh/ruff/pull/9411))
```
Comment (AlexWaygood) [inline CHANGELOG.md:21]: Looks like these are both changes to RUF011 -- is it correct to classify them as flake8-bugbear?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,57 @@
...
+
+- Hug multiline-strings in preview style ([#9243](https://github.com/astral-sh/ruff/pull/9243))
+- [\`flake8-bandit\`] Add `ssl-with-no-version` (`S504`) ([#9384](https://github.com/astral-sh/ruff/pull/9384))
+- [\`flake8-bandit\`] Implement `ssl-insecure-version` (`S502`) ([#9390](https://github.com/astral-sh/ruff/pull/9390))
+- [\`flake8-bandit\`] Implement `ssl-with-bad-defaults` (`S503`) ([#9391](https://github.com/astral-sh/ruff/pull/9391))
+- [\`flake8-bandit\`] Implement suspicious import rules (`S4XX`) ([#8831](https://github.com/astral-sh/ruff/pull/8831))
+- [\`flake8-simplify\`] Implement `zip-dict-keys-and-values` (`SIM911`) ([#9460](https://github.com/astral-sh/ruff/pull/9460))
+- [\`pyflakes\`] Add a fix for `redefined-while-unused` (`F811`) ([#9419](https://github.com/astral-sh/ruff/pull/9419))
+- [\`pylint\`] Implement `unnecessary-dunder-call` (`C2801`) ([#9166](https://github.com/astral-sh/ruff/pull/9166))
+- [\`ruff\`] Add `parenthesize-changed-operators` (`RUF021`) to enforce parentheses in `a or b and c` ([#9440](https://github.com/astral-sh/ruff/pull/9440))
```
Comment (AlexWaygood) [inline CHANGELOG.md:15]:
```suggestion
- [\`ruff\`] Add `parenthesize-chained-operators` (`RUF021`) to enforce parentheses in `a or b and c` ([#9440](https://github.com/astral-sh/ruff/pull/9440))
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,57 @@
...
+### Rule changes
+
+- [\`flake8-boolean-trap\`] Allow Boolean positionals in setters ([#9429](https://github.com/astral-sh/ruff/pull/9429))
+- [\`flake8-bugbear\`] Include subscripts and attributes in static key rule ([#9416](https://github.com/astral-sh/ruff/pull/9416))
+- [\`flake8-bugbear\`] Support variable keys in static dictionary key rule ([#9411](https://github.com/astral-sh/ruff/pull/9411))
+- [\`flake8-builtins\`] Restrict `builtin-attribute-shadowing` (`A003`) to actual shadowed references ([#9462](https://github.com/astral-sh/ruff/pull/9462))
+- [\`flake8-pyi\`] Add fix for `generator-return-from-iter-method` (`PYI058`) ([#9355](https://github.com/astral-sh/ruff/pull/9355))
+- [\`pyflakes\`] Don't flag `redefined-while-unused` (`F811`) in `if` branches ([#9418](https://github.com/astral-sh/ruff/pull/9418))
+- [\`pyupgrade\`] Add some additional Python 3.12 typing members to `deprecated-import` ([#9445](https://github.com/astral-sh/ruff/pull/9445))
+- [\`ruff\`] Add fix for `parenthesize-changed-operators` (`RUF021`) ([#9449](https://github.com/astral-sh/ruff/pull/9449))
```
Comment (AlexWaygood) [inline CHANGELOG.md:26]:
```suggestion
- [\`ruff\`] Add fix for `parenthesize-chained-operators` (`RUF021`) ([#9449](https://github.com/astral-sh/ruff/pull/9449))
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,57 @@
...
+- Disambiguate argument descriptors from section headers ([#9427](https://github.com/astral-sh/ruff/pull/9427))
+- Handle raises with implicit alternate branches ([#9377](https://github.com/astral-sh/ruff/pull/9377))
+- Ignore trailing quotes for unclosed l-brace errors ([#9388](https://github.com/astral-sh/ruff/pull/9388))
+- Respect multi-segment submodule imports when resolving qualified names ([#9382](https://github.com/astral-sh/ruff/pull/9382))
+- Use `DisplayParseError` for stdin parser errors ([#9409](https://github.com/astral-sh/ruff/pull/9409))
+- Use `comment_ranges` for isort directive extraction ([#9414](https://github.com/astral-sh/ruff/pull/9414))
+- Use transformed source code for diagnostic locations ([#9408](https://github.com/astral-sh/ruff/pull/9408))
+- [\`flake8-pyi\`] Exclude `warnings.deprecated` and `typing_extensions.deprecated` arguments ([#9423](https://github.com/astral-sh/ruff/pull/9423))
+- [\`flake8-pyi\`] Fix false negative for `	unused-private-protocol` (`PYI046`) with unused generic protocols ([#9405](https://github.com/astral-sh/ruff/pull/9405))
```
Comment (AlexWaygood) [inline CHANGELOG.md:47]:
```suggestion
- [\`flake8-pyi\`] Fix false negative for `unused-private-protocol` (`PYI046`) with unused generic protocols ([#9405](https://github.com/astral-sh/ruff/pull/9405))
```

**Review by AlexWaygood — COMMENTED** [review]
A few small suggestions ;)

File: CHANGELOG.md (hunk 1 above)
Comment (charliermarsh) [reply CHANGELOG.md:21]: Oops, thanks. This rule was moved to flake8-bugbear, but that PR hasn't actually merged yet, my mistake!

**Comment by charliermarsh** [comment]
Thanks @AlexWaygood, sloppy work on my part, appreciate it :)

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,57 @@
...
+### Formatter
+
+- Generate deterministic IDs when formatting notebooks ([#9359](https://github.com/astral-sh/ruff/pull/9359))
+- Allow `# fmt: skip` with interspersed same-line comments ([#9395](https://github.com/astral-sh/ruff/pull/9395))
+- Parenthesize breaking named expressions in match guards ([#9396](https://github.com/astral-sh/ruff/pull/9396))
+
+### Bug fixes
+
+- Add cell indexes to all diagnostics ([#9387](https://github.com/astral-sh/ruff/pull/9387))
+- Allow `Hashable = None` in type annotations ([#9442](https://github.com/astral-sh/ruff/pull/9442))
```
Comment (AlexWaygood) [inline CHANGELOG.md:37]:
```suggestion
- \[`ruff`\] Allow `Hashable = None` in type annotations (`RUF013`) ([#9442](https://github.com/astral-sh/ruff/pull/9442))
```

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,57 @@
...
+- Avoid infinite loop in constant vs. `None` comparisons ([#9376](https://github.com/astral-sh/ruff/pull/9376))
+- Disambiguate argument descriptors from section headers ([#9427](https://github.com/astral-sh/ruff/pull/9427))
```
Comment (AlexWaygood) [inline CHANGELOG.md:39]:
```suggestion
- [`pydocstyle`] Disambiguate argument descriptors from section headers ([#9427](https://github.com/astral-sh/ruff/pull/9427))
```

**Review by zanieb — APPROVED** [review]
<3

**Review by AlexWaygood — APPROVED** [review]
Looks great! Hyped to use some of these in my projects

**Pull request merged**